### DISCUSSION TITLE
How to add 2 LevelControl in one Endpoint

### DISCUSSION THREAD

**Discussion by MatthieuCa** [matter-js/matter.js#1000] [Q&A] [answered]
Hello!

I am playing around with matter. js but I am facing a problem. I want to connect a device that needs a control of the speed and the dimmer.

I was wondering if someone knows how to implement a double Level Control on a device?

Thank you
Matthieu

**Comment by JimBuzbee** [comment, marked as answer]
Take a look at the ComposedDevice example that shows how to combine capabilities into a single device. But one big question will be how controllers like Google, Apple, Home Assistant, Alexa, etc, will handle it. They may just ignore the second level control.

**Reply by Apollon77** [reply]
Usually they understand this.
Apple shows by default two sliders in one device screen - and you can split it on different tiles. Google will  only show it as two tiles. But yes: Simply add multiple endpoints - each with a LevelControl cluster

**Reply by MatthieuCa** [reply]
If I understand well, I need to create 2 endpoints to have a second LevelControl ? 

Because the problem I have when I create 2 endpoints is that on Google Home, the device is split in half like in this screenshot (The exemple is based on DeviceNode.ts

![Screenshot_20240708_144818_Home](https://github.com/project-chip/matter.js/assets/174323286/80e19676-5968-4c31-a902-5dc274025185)

So I have the 2 LevelControl but not on the same device...

**Reply by MatthieuCa** [reply]
> Usually they understand this. Apple shows by default two sliders in one device screen - and you can split it on different tiles. Google will only show it as two tiles. But yes: Simply add multiple endpoints - each with a LevelControl cluster

Ok thank you so don't take into count my last message. Just it is possible to add Clusters to Endpoint or it's only Device ?

**Reply by Apollon77** [reply]
In fact to be exact:

You define "a" Device type for your endpoint from the list of available device types, "a" is not 100 % always 1 because it can be multiple application device types WHEN they have a superset/subset relationship (e.g. enhanced colorlight can have dimmable light and also OnOffLight types). Additionally there are some utility device types e.g. PowerSource) allowed to be added. 

BUT it is not "freely", so adding a temperature sensor and a humidity sensor device type on one endpoint is not allowed because these have no such described relation and need to go to different endpoints.

On the other side you can always add additional cluster servers to an endpoint, but here noone defines and knows what will happen, if they are shown, ignored or what happens.

So solution for you is to have two endpoints with the levelcontrol cluster and "Dimmable" devicetypes .

In fact "all together" (unless you start with Bridges/Aggregators) you have a simple device and in this case a composed device
